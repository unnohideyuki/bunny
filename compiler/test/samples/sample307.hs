x = 1.0 :: Double
y = 2.0 :: Double
z = 0.0 :: Double
t = 10.0 :: Double

main = do print $ succ (2.3 :: Double)
          print $ pred (2.3 :: Double)
          print $ (toEnum 3 :: Double)
          print $ fromEnum (2.3 :: Double)
          print $ take 10 $ [x..]
          print $ take 10 $ [y, x ..]
          print $ [z .. t]
          print $ [z, y .. t]
