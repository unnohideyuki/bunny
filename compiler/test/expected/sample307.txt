3.3
1.2999999999999998
3.0
2
[1.0,2.0,3.0,4.0,5.0,6.0,7.0,8.0,9.0,10.0]
[2.0,1.0,0.0,-1.0,-2.0,-3.0,-4.0,-5.0,-6.0,-7.0]
[0.0,1.0,2.0,3.0,4.0,5.0,6.0,7.0,8.0,9.0,10.0]
[0.0,2.0,4.0,6.0,8.0,10.0]
